pkg/cluster-runtime/installer.go

```diff
@@ -148,8 +149,14 @@ func (i *Installer) Install() error {
 	}
 
 	// distribute rootfs
-	if err := i.Distributor.Distribute(all, rootfs); err != nil {
-		return err
+	if i.P2PDistributor != nil {
```

Comment:
I wonder if we use distibutor as the interface and initialize different instances, there's no need to make an if else decision here

Reply:
This is a temporary solution. In the future, I plan to introduce multicast distribution as well, so multiple distributors will be running in one cluster. SFTP-based distributor will be the fallback method (when other methods won't work due to environmental limitations or failure).

Reply:
Even so, we should solve this problem at the implementation level, not at the use level.

maybe we could have an intelligent distributor implementation that invokes several other distributors and have the ability to fallback.

Reply:
> Even so, we should solve this problem at the implementation level, not at the use level.
> 
> maybe we could have an intelligent distributor implementation that invokes several other distributors and have the ability to fallback.

Okay, I'll unify the two distributors here for now.

Reply:
Done. Is there anything else that I need to pay attention to? If no I'll be squashing my commits shortly.

Reply:
just do it 😄